hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1843,20 +1852,144 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	return result, nil
 }
 
+// annotationsForCertRenewal returns a set of annotations to set based on the current state of the KAS
+// serving certificate. These could include the annotation to restart control plane pods.
+// It will always return an empty map whenever the HostedCluster should not be checked for
+// stale certs (see shouldCheckForStaleCerts)
+func annotationsForCertRenewal(
+	log logr.Logger,
+	hcp *hyperv1.HostedControlPlane,
+	shouldCheckForStaleCerts func() bool,
+	kasServingCertHashFromSecret func() (string, error),
+	kasServingCertHashFromEndpoint func() (string, error)) func() (map[string]string, error) {
+	return func() (map[string]string, error) {
+		if !shouldCheckForStaleCerts() {
+			return nil, nil
+		}
+
+		// The hash from the KAS serving secret is the source of truth for what
+		// certificate we should be using for KAS
+		// We can check whether we need to restart pods in one of 2 ways:
+		// 1. We have previously saved an annotation on the HCP that says which was the last
```

Comment:
nit: 1 also uses the annotation to restart but reading this comment is not clear.